crates/blockchain-tree/src/blockchain_tree.rs

```diff
@@ -323,8 +323,20 @@ impl<DB: Database, C: Consensus, EF: ExecutorFactory> BlockchainTree<DB, C, EF>
         }
 
         // insert block inside unconnected block buffer. Delaying its execution.
-        self.buffered_blocks.insert_block(block);
-        Ok(BlockStatus::Disconnected)
+        self.buffered_blocks.insert_block(block.clone());
+
+        // find the lowest ancestor of the block in the buffer to return as the missing parent
+        // this shouldn't return None because that only happens if the block was evicted, which
+        // shouldn't happen right after insertion
+        let lowest_ancestor =
+            self.buffered_blocks.lowest_ancestor(&block.hash).ok_or_else(|| {
```

Comment:
it's a bit confusing that `lowest_ancestor` method can return the block itself. can we add a doc or make the method name clearer?

Reply:
Most of the method uses are made simpler if the method can return the block itself, so i feel like we should keep the functionality, although I agree that it's kind of confusing since a block is not its own ancestor / parent. Just added more docs, a new name would be great but I couldn't really think of anything